src/Raven.Server/Documents/PeriodicBackup/BackupConfigurationHelper.cs

```diff
@@ -39,6 +43,27 @@ public static void UpdateLocalPathIfNeeded(PeriodicBackupConfiguration configura
             configuration.LocalSettings.FolderPath = pathResult.FolderPath;
         }
 
+        public static void UpdateExcludedDatabasesIfNeeded(ServerWideBackupConfiguration configuration, ServerStore serverStore)
+        {
+            if (configuration.BackupType != BackupType.Snapshot)
+                return;
+
+            using (serverStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var excludes = new HashSet<string>(configuration.ExcludedDatabases, StringComparer.OrdinalIgnoreCase);
+                var dbs = serverStore.Cluster.GetAllRawDatabases(context);
+
+                foreach (var db in dbs)
+                {
+                    if (db.IsSharded)
+                        excludes.Add(db.DatabaseName);
+                }
+
+                configuration.ExcludedDatabases = excludes.ToArray();
+            }
+        }
```

Comment:
Do we want to move this into the cluster state machine? to avoid potential race between creating a database and server-side backup config.

Reply:
done, moved this to cluster state machine